### PR TITLE
Feature: Add Remote Modal Widget

### PR DESCRIPTION
**Pull request by yemexx1**

### CODE REVIEW

**Review by netwox — APPROVED** [review]

File: src/Assets/AppAsset.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
```
Comment (olajideoye) [inline src/Assets/AppAsset.php]: @yemexx1  I don't think this asset is needed here. In reality, the app asset definition could vary according to the project.

File: src/Assets/AppModuleAsset.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * @package app\assets
+ */
+class AppModuleAsset extends LocalAssetBundle
+{
+    public $js = [
+        'js/app.js'
+    ];
+
+    public $depends = [
+        'CottaCush\Yii2\Assets\AppAsset'
```
Comment (olajideoye) [inline src/Assets/AppModuleAsset.php]: @yemexx1 The dependency here should be the `yii\web\JqueryAsset`

File: src/Assets/AppModuleAsset.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace CottaCush\Yii2\Assets;
+
+/**
+ * Class AppModuleAsset
+ * @author Olajide Oye <jide@cottacush.com>
```
Comment (olajideoye) [inline src/Assets/AppModuleAsset.php]: @yemexx1 Please add yourself as the author :)

File: src/Widgets/BaseIcon.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?php
```
Comment (olajideoye) [inline src/Widgets/BaseIcon.php]: @yemexx1 The use for this widget has been replaced with a static function in the `CottaCush\Yii2\Helpers\Html` class.

File: src/Widgets/BaseRemoteModalWidget.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace CottaCush\Yii2\Widgets;
+
+use CottaCush\Yii2\Assets\ModalAsset;
+use yii\base\Widget;
+use yii\bootstrap\Html;
```
Comment (olajideoye) [inline src/Widgets/BaseRemoteModalWidget.php]: @yemexx1 Could we use the `CottaCush\Yii2\Helpers\Html` class instead?

File: src/Widgets/FontAwesomeIcon.php [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (olajideoye) [inline src/Widgets/FontAwesomeIcon.php]: @yemexx1 The use of this widget has been replaced with a static function on the `CottaCush\Yii2\Helpers\Html` class

File: src/Widgets/RemoteModalFormWidget.php [hunk 7]
```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace CottaCush\Yii2\Widgets;
+
+use yii\helpers\Html;
```
Comment (olajideoye) [inline src/Widgets/RemoteModalFormWidget.php]: Please replace this class with the `CottaCush\Yii2\Helpers\Html` class

File: src/Widgets/RemoteModalPlaceholderWidget.php [hunk 8]
```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace CottaCush\Yii2\Widgets;
+
+use CottaCush\Yii2\Assets\ModalAsset;
+use yii\bootstrap\Modal;
+use yii\helpers\Html;
```
Comment (olajideoye) [inline src/Widgets/RemoteModalPlaceholderWidget.php]: Please use the `CottaCush\Yii2\Helpers\Html` class

File: src/Widgets/RemoteModalWidget.php [hunk 9]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace CottaCush\Yii2\Widgets;
+
+use yii\helpers\Html;
```
Comment (olajideoye) [inline src/Widgets/RemoteModalWidget.php]: @yemexx1 Please use the `CottaCush\Yii2\Helpers\Html` class

**Review by olajideoye — CHANGES_REQUESTED** [review]
There're a few things to clean up

**Review by goke-epapa — APPROVED** [review]

**Review by olajideoye — APPROVED** [review]
Looks good to merge!

**Pull request merged**